data-prepper-plugins/service-map-stateful/src/main/java/org/opensearch/dataprepper/plugins/processor/ServiceMapStatefulProcessor.java

```diff
@@ -289,8 +303,26 @@ public void shutdown() {
         currentWindow.delete();
         previousTraceGroupWindow.delete();
         currentTraceGroupWindow.delete();
+        previousIsolatedServiceNodes.clear();
+        currentIsolatedServiceNodes.clear();
     }
 
+    private void processIsolatedServiceMapNodes(final Collection<Record<Event>> serviceDependencyRecords) {
+        LOG.debug("Add isolated service nodes into service-map relationships.");
```

Comment:
These might be more appropriate for the TRACE log level, but I'm ok with either.